crates/core/src/polynomial/arith_circuit.rs

```diff
@@ -10,51 +10,103 @@ use stackalloc::{
 	stackalloc_uninit,
 };
 
+/// Convert the expression to a sequence of arithmetic operations that can be evaluated in sequence.
 /// Convert the expression to a sequence of arithmetic operations that can be evaluated in sequence.
 fn circuit_steps_for_expr<F: Field>(
```

Comment:
We should change this to convert `ArithCircuit` to `ArithCircuitPoly` instead of converting from `ArithExpr`. `ArithExpr` -> `ArithCircuit` -> `ArithCircuitPoly`.